src/NewtonKrylov.fypp

```diff
@@ -0,0 +1,418 @@
+#:include "../include/common.fypp"
+#:set RC_KINDS_TYPES = REAL_KINDS_TYPES + CMPLX_KINDS_TYPES
+module LightKrylov_NewtonKrylov
+   use stdlib_optval, only: optval
+   use LightKrylov_Constants
+   use LightKrylov_Logger
+   use LightKrylov_AbstractVectors
+   use LightKrylov_AbstractLinops
+   use LightKrylov_AbstractSystems
+   use LightKrylov_IterativeSolvers
+   use LightKrylov_Utils
+
+   implicit none
+   private
+
+   character*128, parameter :: this_module = 'LightKrylov_NewtonKrylov'
```

Comment:
Use `character(len=128)` to be consistent with the convention used in the other modules.